sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -656,6 +656,8 @@ abstract class TypeCoercionBase {
       (leftType, rightType) match {
         case (_: DecimalType, NullType) => true
         case (NullType, _: DecimalType) => true
+        case (_: DayTimeIntervalType, _: DayTimeIntervalType) => true
+        case (_: YearMonthIntervalType, _: YearMonthIntervalType) => true
```

Comment:
why do we need this? the default case should work for interval types

Reply:
Removed